[ci] kill PyTorch py2 circle jobs

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29400 test py2 removal on nightlies
* #29357 [ci] remove py2 flake8 lint
* **#29353 [ci] kill PyTorch py2 circle jobs**

First step to killing Python 2 everywhere. I don't really know that much
about the caffe2 circle jobs so I left them alone for now.

Differential Revision: [D18920563](https://our.internmc.facebook.com/intern/diff/D18920563)

suo committed:
[ci] kill PyTorch py2 circle jobs

First step to killing Python 2 everywhere. I don't really know that much
about the caffe2 circle jobs so I left them alone for now.

suo commented:
@ezyang What's the best way to test the binary build/test/upload stuff?

kostmo commented:
## CircleCI build failures summary
As of commit c2f405a:
* 1/3 recognized as flaky
* 2/3 broken upstream. You may want to rebase on the [latest viable branch](https://dr.pytorch.org/master-viable-commits.html).
* 1/3 failures introduced in this PR



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=c2f405aa7236eed139946abbfe3f8cae096ef760).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

ezyang commented:
> What's the best way to test the binary build/test/upload stuff?

Manually trigger it on PR by editing the triggers. I think in the latest iteration, you just need to edit this code: https://github.com/pytorch/pytorch/blob/8498a1555b16155859a1de011786db7bb6f3012a/.circleci/cimodel/data/binary_build_definitions.py#L66

suo committed:
Update on "[ci] kill PyTorch py2 circle jobs"


First step to killing Python 2 everywhere. I don't really know that much
about the caffe2 circle jobs so I left them alone for now.

suo committed:
Update on "[ci] kill PyTorch py2 circle jobs"


First step to killing Python 2 everywhere. I don't really know that much
about the caffe2 circle jobs so I left them alone for now.
